velox/functions/prestosql/ArraysOverlap.cpp

```diff
@@ -0,0 +1,160 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/lib/LambdaFunctionUtil.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+template <typename T>
+class ArraysOverlapFunction : public exec::VectorFunction {
+ public:
+  ArraysOverlapFunction() {}
+
+  explicit ArraysOverlapFunction(
+      SetWithNull<T> constantSet,
+      bool isLeftConstant)
+      : constantSet_(std::move(constantSet)), isLeftConstant_(isLeftConstant) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    // if one of them is a ConstantVector.
+    // Iterate through the other ArrayVector->elementsFlatVector
+    // and do constantSet_.find()
+    BaseVector* left = args[0].get();
+    BaseVector* right = args[1].get();
+    if (constantSet_.has_value() && isLeftConstant_) {
+      std::swap(left, right);
+    }
+    exec::LocalDecodedVector decoder(context, *left, rows);
+    exec::LocalDecodedVector elementsDecoder(context);
+    auto decodedLeftElements =
+        getDecodedElementsFromArrayVector(decoder, elementsDecoder, rows);
+
+    FlatVector<bool>* resBoolVector = (*result)->asFlatVector<bool>();
+    auto processRow = [&](const vector_size_t rowId,
+                          const SetWithNull<T>& rightSet) {
+      auto decodedLeftArray = decoder.get();
+      auto baseLeftArray = decodedLeftArray->base()->as<ArrayVector>();
+      auto idx = decodedLeftArray->index(rowId);
+      auto offset = baseLeftArray->offsetAt(idx);
+      auto size = baseLeftArray->sizeAt(idx);
+
+      for (auto i = offset; i < (offset + size); ++i) {
+        // For each element in the current row search for it in the rightSet.
+        if (decodedLeftElements->isNullAt(i)) {
+          continue;
+        }
+        if (rightSet.set.count(decodedLeftElements->valueAt<T>(i)) > 0) {
+          // Found and overlapping element. Add to result set.
+          resBoolVector->set(rowId, true);
+          return;
+        }
+      }
+      // If none of them is found in the rightSet, set false for current row
+      // indicating there are no overlapping elements with rightSet.
+      resBoolVector->set(rowId, false);
+    };
+
+    if (constantSet_.has_value()) {
+      rows.applyToSelected(
+          [&](vector_size_t row) { processRow(row, *constantSet_); });
+    }
+    // General case when no arrays are constant and both sets need to be
+    // computed for each row.
+    else {
+      exec::LocalDecodedVector rightDecoder(context, *right, rows);
```

Comment:
You can use getDecodedElementsFromArrayVector for right side also here I think. Similar for ArraysIntersectExcept.